eth-node/types/waku.go

```diff
@@ -122,6 +122,8 @@ type Waku interface {
 
 	SubscribeToConnStatusChanges() (*ConnStatusSubscription, error)
 
+	CheckForMissingMessages(peerID peer.ID, pubsubTopic string, contentTopics []string) error
```

Comment:
Please correct me if I am wrong, but if `CheckForMissingMessages` is called in a row for the same `pubsubTopic`, then the `contentTopics` of the former call will most likely not be fetched.

Let's consider following scenario:
```golang
t.CheckForMissingMessages(peerID, defaultPubsubTopic, []string{"A", "B"})
...
t.CheckForMissingMessages(peerID, defaultPubsubTopic, []string{"C", "D", "E"})
```

Will the content topics "A" and "B" be fetched?

Reply:
They will not be fetched. But it's okay because CheckForMissingMessages is called with all the filters that are active. If you stop receiving a filter, is because you're no longer subscribed to it, thus no need to ask for messages from that content topic. 

This scenario can happen for example if you leave a community.

Reply:
Got it.

To me, it breaks the principle of least surprise 😅 The `CheckForMissingMessages` API itself is not clear, as it requires an understanding of the implementation. Perhaps it would be worth adding a description or improving the name (`SetContentTopicsToBePeriodicallyCheckedForMissingMessages` :kek:).

Reply:
That function name is perfect. Will use your suggestion :rocket: